assets/js/components/product-control/index.js

```diff
@@ -252,70 +143,93 @@ class ProductControl extends Component {
 				{ ...a11yProps }
 			/>
 		);
-	}
+	};
 
-	render() {
-		const { products, loading, product, variationsList } = this.state;
-		const { onChange, renderItem, selected } = this.props;
-		const currentVariations = variationsList[ product ] || [];
-		const currentList = [ ...products, ...currentVariations ];
-		const messages = {
-			list: __( 'Products', 'woo-gutenberg-products-block' ),
-			noItems: __(
-				"Your store doesn't have any products.",
-				'woo-gutenberg-products-block'
-			),
-			search: __(
-				'Search for a product to display',
-				'woo-gutenberg-products-block'
-			),
-			updated: __(
-				'Product search results updated.',
-				'woo-gutenberg-products-block'
-			),
-		};
-		const selectedListItems = selected
-			? [ find( currentList, { id: selected } ) ]
-			: [];
+	const getRenderItemFunc = () => {
+		if ( renderItem ) {
+			return renderItem;
```

Comment:
We would need to recreate the function on every re-render if props change anyway since it relies on some of them, no? So there would be no benefit.

Reply:
Right, (I've brought this up elsewhere as well I think).  I'm just wary of getting used to a pattern where it may _not_ be the case or future refactors mean the props are less likely to change.  We can leave this, it's a pattern that I will probably still note in future reviews :)